pkg/maps/ctmap/ctmap.go

```diff
@@ -239,19 +245,28 @@ func doGC6(m *bpf.Map, filter *GCFilter) int {
 			del = true
 			//log.WithField(logfields.Object, logfields.Repr(entry)).Debug("Deleting IPv4 entry since it timeout")
 		}
+
+		n := nextKey.ToHost().(*CtKey6Global)
 		if filter.fType&GCFilterByID != 0 &&
 			// In CT's entries, saddr is the packet's receiver,
 			// which means, is the destination container IP.
-			nextKey.saddr.IP().Equal(filter.IP) {
+			n.saddr.IP().Equal(filter.IP) {
+
+			nipp := policy.NIPortProto{
+				SecID: policy.NumericIdentity(entry.src_sec_id),
+				Port:  n.sport,
+				Proto: uint8(n.nexthdr),
+			}
 
 			// Check if the src_sec_id of that entry is not allowed
 			// to talk with the destination container IP.
-			if _, ok := filter.IDsToRm[entry.src_sec_id]; ok {
-
+			if _, ok := filter.IDsToRm[nipp]; ok {
 				del = true
 				//log.WithFields(log.Fields{
```

Comment:
This comment should be deleted if it isn't being used.

Reply:
Fixed